src/applications/check-in/day-of/pages/LoadingPage/index.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useCallback, useEffect, useMemo } from 'react';
+import { useDispatch, useSelector, batch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { URLS } from '../../../utils/navigation/day-of';
+import { useFormRouting } from '../../../hooks/useFormRouting';
+import {
+  receivedEmergencyContact,
+  receivedDemographicsData,
+  receivedNextOfKinData,
+  receivedMultipleAppointmentDetails,
+  triggerRefresh,
+  updateFormAction,
+} from '../../../actions/day-of';
+import { useSessionStorage } from '../../../hooks/useSessionStorage';
+import { makeSelectCheckInData } from '../../hooks/selectors';
+import { api } from '../../../api';
+
+const LoadingPage = props => {
+  const {
+    isSessionLoading,
+    router,
+    isEmergencyContactEnabled,
+    checkInExperienceUpdateInformationPageEnabled,
+  } = props;
+  const { goToErrorPage, goToNextPage } = useFormRouting(router, URLS);
+  const selectCheckInData = useMemo(makeSelectCheckInData, []);
+  const checkInData = useSelector(selectCheckInData);
+  const { getCurrentToken } = useSessionStorage(false);
+  const { context, appointments } = checkInData;
+  const dispatch = useDispatch();
+  const setSessionData = useCallback(
+    (payload, token) => {
+      batch(() => {
+        const {
+          appointments: appts,
+          demographics: demo,
+          patientDemographicsStatus,
+        } = payload;
+        dispatch(triggerRefresh(false));
+        dispatch(receivedMultipleAppointmentDetails(appts, token));
+        dispatch(
+          updateFormAction({
+            patientDemographicsStatus,
+            isEmergencyContactEnabled,
+            checkInExperienceUpdateInformationPageEnabled,
+          }),
+        );
+        if (typeof demo !== 'undefined') {
+          dispatch(receivedDemographicsData(demo));
```

Comment:
Is this now the time to combine the `receievedXXXXX` dispatches?

Reply:
Yeah, I was thinking about that too. I think Ideally we could have one dispatch for all of the payload kind of like how pre-check-in is doing it. Good stuff for the details on the next ticket.

Reply:
Add this as a note in the description of https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/34902